src/components/TodoList.tsx

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import { Todo as TodoType } from '../types/Todo';
+import { TodoListItem } from './TodoListItem';
+
+type Props = {
+  filteredTodos: TodoType[];
+  onDelete: (id: number) => void;
+  isLoading: boolean;
+  tempTodo: TodoType | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  filteredTodos,
+  onDelete,
+  isLoading,
+  tempTodo,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      <TransitionGroup>
+        {filteredTodos.map(({ id, title, completed }) => (
+          <CSSTransition key={id} timeout={300} classNames="item">
+            <TodoListItem
+              id={id}
+              title={title}
+              completed={completed}
+              onDelete={onDelete}
+              isLoading={isLoading}
+            />
+          </CSSTransition>
+        ))}
+
+        {tempTodo && (
+          <CSSTransition key={tempTodo.id} timeout={300} classNames="item">
```

Comment:
there should be different class for temp todo animation
```suggestion
          <CSSTransition key={tempTodo.id} timeout={300} classNames="temp-item">
```